### PR TITLE
prov/efa: fix an uninitialized variable in unit test

### PR DESCRIPTION
**Pull request by wenduwan**
This patch fixes a compiler warning

### CODE REVIEW

**Review by shijin-aws — APPROVED** [review]

File: prov/efa/test/efa_unit_test_ep.c [hunk 1]
```diff
@@ -10,7 +10,7 @@
  */
 void test_rxr_ep_host_id(struct efa_resource **state, bool file_exists, char *raw_id, uint64_t expect_id)
 {
-	int fd;
+	int fd = -1;
```
Comment (lrbison) [inline prov/efa/test/efa_unit_test_ep.c:13]: Does the compiler complain if line 41 depends on `file_exists` I directly?  It seems like the static analysis isn't quite putting together than file_exists implies rxr_env.host_id_file

**Review by lrbison — APPROVED** [review]

File: prov/efa/test/efa_unit_test_ep.c (hunk 1 above)
Comment (wenduwan) [reply prov/efa/test/efa_unit_test_ep.c:13]: > Does the compiler complain if line 41 depends on file_exists I directly? It seems like the static analysis isn't quite putting together than file_exists implies rxr_env.host_id_file

It also works.

**Comment by wenduwan** [comment]
bot:aws:retest

**Comment by shijin-aws** [comment]
bot:aws:retest

**Comment by wenduwan** [comment]
@shijin-aws AWS CI passed. Ready to merge.

**Pull request merged**